File: Master/1st Course/Fall semester/Paralelie procesi un valoda ADA/Lab8/Task1/C#/MatrFun.cs
using System;
using System.Threading;
using System.Runtime.CompilerServices;

namespace ThreadingTask {

    class MatrFun {
        [MethodImpl(MethodImplOptions.Synchronized)]
        public static void PrintRow(int row, int[,] matrix) {
            // Outputting current row.
            Console.Write("Row: {0}. Elements: ", row);

            for (int j = 0; j < matrix.GetLength(1); j++) {
                // Getting row elements.
                Console.Write(matrix[row, j] + " ");
                Thread.Sleep(0);
            }

            Console.WriteLine();
        }
    }
}

File: Master/1st Course/Fall semester/Paralelie procesi un valoda ADA/Lab8/Task1/Java/MatrFun.java
package lab8.threading;

public class MatrFun {	
    synchronized static void printRow(int row, int[][] matrix) {
        System.out.print("Row: " + row + ". Elements: ");
        
        for (int j = 0; j < matrix[row].length; j++) {
            System.out.print(matrix[row][j] + " ");
            
            try {
                Thread.sleep(0);
            }
            catch(InterruptedException e) {
                System.out.println(e.getMessage());
            }
        }
        
        System.out.println();
    }
}